libvmi/driver/kvm/kvm.c

```diff
@@ -1115,11 +1115,12 @@ kvm_get_memory_patch(
     }
     else {
         // get the data from kvm
-        nbytes =
-            read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
-        if (nbytes != (length + 1)) {
+        nbytes = read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
+        if ( nbytes <= 0 )
```

Comment:
What if nbytes = length/2 ? Was the original condition intended to check that data of proper size was returned, or just for an invalid rc from read()?